src/core/Makefile.am

```diff
@@ -227,13 +226,10 @@ HEADERS_COMMON                      = \
     crypto/pbkdf2_cmac.h              \
     crypto/sha256.hpp                 \
     mac/mac.hpp                       \
-    mac/mac_blacklist.hpp             \
-    mac/mac_blacklist_impl.hpp        \
-    mac/mac_blacklist_stub.hpp        \
+    mac/mac_filter.hpp                \
+    mac/mac_filter_impl.hpp           \
+    mac/mac_filter_stub.hpp           \
```

Comment:
What about removing this stub implementation and simply wrapping all calls to the mac filter using `OPENTHREAD_ENABLE_MAC_FILTER`?